### ISSUE TITLE
lookup password module with path '/dev/null' returns empty string

### ISSUE THREAD

**Issue by yabhinav** [ansible/ansible#22359]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
lookup

##### ANSIBLE VERSION
```
[user@server ]$ ansible --version
ansible 2.2.1.0
  config file = /home/ansible.cfg
  configured module search path = ['./library']
```

##### CONFIGURATION
```
# some basic default values...
[defaults]
inventory       = ./inventory/hosts
ansible.verbose = "v"
library         = ./library
remote_tmp      = /tmp/$USER/.ansible
local_tmp       = /tmp/$USER/.ansible
pattern         = *
forks           = 5
poll_interval   = 15
transport       = smart
remote_port     = 22
module_lang     = C
scp_if_ssh      = True
gathering = implicit
roles_path    = ./roles
host_key_checking = False
sudo_user       = root
ask_sudo_pass   = False
ask_pass        = False
sudo_exe = sudo
timeout = 10
remote_user = root
log_path = ./ansible.log
module_name = shell
executable = /bin/bash

# set plugin path directories here, separate with colons
action_plugins     = ~/.ansible/plugins/action_plugins/:/usr/share/ansible_plugins/action_plugins:./action_plugins/
callback_plugins   = ~/.ansible/plugins/callback_plugins/:/usr/share/ansible_plugins/callback_plugins:./callback_plugins/
connection_plugins = ~/.ansible/plugins/connection_plugins/:/usr/share/ansible_plugins/connection_plugins:./connection_plugins/
lookup_plugins     = ~/.ansible/plugins/lookup_plugins/:/usr/share/ansible_plugins/lookup_plugins:./lookup_plugins/
vars_plugins       = ~/.ansible/plugins/vars_plugins/:/usr/share/ansible_plugins/vars_plugins:./vars_plugins/
filter_plugins     = ~/.ansible/plugins/filter_plugins/:/usr/share/ansible_plugins/filter_plugins:./filter_plugins/
nocows = 0
ANSIBLE_COW_SELECTION=random
http_user_agent = ansible-agent
fact_caching = memory

retry_files_save_path = ~/.ansible-retry

[privilege_escalation]
#become=True
become_method='dzdo'
#become_user='root'
#become_ask_pass=False
[ssh_connection]
control_path = ~/.ssh/ansible-%%r@%%h:%%p
[accelerate]
accelerate_port = 5099
accelerate_timeout = 30
accelerate_connect_timeout = 5.0
```

##### OS / ENVIRONMENT
Redhat Linux 6.5

##### SUMMARY
When using [lookup-password module](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/lookup/password.py) to generate a random password, as mentioned by documentation [here](http://docs.ansible.com/ansible/playbooks_lookups.html) using `/dev/null` should generate the password string , also verified in the module code that same thing is happening, but when  running the same in a task an empty string is generated probably because the  module is looking for /dev/null file and it is empty. 

##### STEPS TO REPRODUCE

```yaml
- debug: 
    msg: "{{ lookup('password', '/dev/null chars=ascii_letters length=15') }}"

- debug: 
    msg: "{{ lookup('password', '/dev/null') }}"
```

##### EXPECTED RESULTS
```
ok: [testlab] => {
    "msg": "trtretreteytreert2324sf"
}
```

##### ACTUAL RESULTS
```
ok: [testlab] => {
    "msg": ""
}
```

This issue might be due to the  failure of the  checks and the module is looking for password in /dev/null

```
if content is None or b_path == to_bytes('/dev/null'):

if changed and b_path != to_bytes('/dev/null'):
```

**Comment by Lujeni** [comment]
Hello,

This bug seems fixed in the devel branch (https://github.com/ansible/ansible/commit/6fae1d2bbfa1086e4fc18e95acac96f98d838480).

**Comment by bcoca** [comment]
closing as per above

**Comment by xv-mikael-k** [comment]
That it is implemented in 2.3 should be noted in the documentation.
